### ISSUE TITLE
xmonad-with-packagages needlessly conflicts with ghc

### ISSUE THREAD

**Issue by anka-213** [NixOS/nixpkgs#60914]
## Issue description
For some reason the man page for ghc is included in the output for xmonad-with-packages, which means that it cannot (easily) be installed together with ghc.

### Steps to reproduce

```
$ nix-env -f '<nixpkgs>' -iA ghc xmonad-with-packages                                                                                   
installing 'ghc-8.6.5'
installing 'xmonad-with-packages-8.6.5'
building '/nix/store/37s37cjsv5kypgpr2xg5q19rgf5bm22b-user-environment.drv'...
error: packages '/nix/store/zghq088rbvlikc2ns6adgmhks423mcin-ghc-8.6.5/share/man/man1/ghc.1.gz' and '/nix/store/rhpizad2ys1c99kgyi1fah1fcqzz81wv-xmonad-with-packages-8.6.5/share/man/man1/ghc.1.gz' have the same priority 5; use 'nix-env --set-flag priority NUMBER INSTALLED_PKGNAME' to change the priority of one of the conflicting packages (0 being the highest priority)
builder for '/nix/store/37s37cjsv5kypgpr2xg5q19rgf5bm22b-user-environment.drv' failed with exit code 1
error: build of '/nix/store/37s37cjsv5kypgpr2xg5q19rgf5bm22b-user-environment.drv' failed
```

## Technical details

```
$ ls -Rl /nix/store/rhpizad2ys1c99kgyi1fah1fcqzz81wv-xmonad-with-packages-8.6.5/                                                           
/nix/store/rhpizad2ys1c99kgyi1fah1fcqzz81wv-xmonad-with-packages-8.6.5/:
total 0
dr-xr-xr-x 1 root root 12  1 jan  1970 bin/
dr-xr-xr-x 1 root root  6  1 jan  1970 share/

/nix/store/rhpizad2ys1c99kgyi1fah1fcqzz81wv-xmonad-with-packages-8.6.5/bin:
total 4
-r-xr-xr-x 1 root root 378  1 jan  1970 xmonad*

/nix/store/rhpizad2ys1c99kgyi1fah1fcqzz81wv-xmonad-with-packages-8.6.5/share:
total 4
lrwxrwxrwx 1 root root 77  1 jan  1970 man -> /nix/store/c27gk5c51hrb60wwgbl751g5rxr3qmpx-ghc-8.6.5-with-packages/share/man/
```
```
$ ls -Rl /nix/store/c27gk5c51hrb60wwgbl751g5rxr3qmpx-ghc-8.6.5-with-packages/share/man/                                                    
/nix/store/c27gk5c51hrb60wwgbl751g5rxr3qmpx-ghc-8.6.5-with-packages/share/man/:
total 0
dr-xr-xr-x 1 root root 38  1 jan  1970 man1/

/nix/store/c27gk5c51hrb60wwgbl751g5rxr3qmpx-ghc-8.6.5-with-packages/share/man/man1:
total 8
lrwxrwxrwx 1 root root 77  1 jan  1970 ghc.1.gz -> /nix/store/zghq088rbvlikc2ns6adgmhks423mcin-ghc-8.6.5/share/man/man1/ghc.1.gz
lrwxrwxrwx 1 root root 82  1 jan  1970 xmonad.1.gz -> /nix/store/k97r4wf1ffmzqf0qii0gky493403ylbx-xmonad-0.15/share/man/man1/xmonad.1.gz
```

```
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.0.7-arch1-1-ARCH, Arch Linux, noversion`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.2.1`
 - channels(root): `"nixpkgs-19.09pre177709.fe41fdc3cbe"`
 - channels(hjulle): `"nixospkgs-19.09pre175916.1fc591f9a5b"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

**Comment by JohnAZoidberg** [comment]
This would fix it:

```diff
--- i/pkgs/applications/window-managers/xmonad/wrapper.nix
+++ w/pkgs/applications/window-managers/xmonad/wrapper.nix
@@ -8,8 +8,7 @@ in stdenv.mkDerivation {
   nativeBuildInputs = [ makeWrapper ];

   buildCommand = ''
-    mkdir -p $out/bin $out/share
-    ln -s ${xmonadEnv}/share/man $out/share/man
+    install -D ${xmonadEnv}/share/man/man1/xmonad.1.gz $out/share/man/man1/xmonad.1.gz
     makeWrapper ${xmonadEnv}/bin/xmonad $out/bin/xmonad \
       --set NIX_GHC "${xmonadEnv}/bin/ghc" \
       --set XMONAD_XMESSAGE "${xmessage}/bin/xmessage"
```

But do we really need to use `ghcWithPackages (self: [ self.xmonad ] ++ packages self);` for this package?
This make ghc a runtime dependency.

So your problem would also be fixed if you install only `xmonad-with-packages` becaues that gives you ghc for free ;)

**Comment by anka-213** [comment]
@JohnAZoidberg Yes, ghc should be a runtime dependency of xmonad, since configuration is done by recompiling the application. But no, ghc is not made available by xmonad-with-packages, just the man-page. :P

**Comment by JohnAZoidberg** [comment]
Hmm, well then I think my patch is the best solution.

**Comment by anka-213** [comment]
@JohnAZoidberg Does `makeWrapper` and `install` automatically create the neccessary directories if they do not exist? If so, the patch looks good to me.

**Comment by JohnAZoidberg** [comment]
Yup

**Comment by JohnAZoidberg** [comment]
Solved in 71fd248.
